pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/utils/argminmax/ArgMinMaxWrapperValSet.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.aggregation.utils.argminmax;
+
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.core.common.BlockValSet;
+
+
+public class ArgMinMaxWrapperValSet {
```

Comment:
is my understanding correct that the comparator functions are only used for measuring columns and the value related functions only used for projection columns? Just wondering if it'll be cleaner to have a base class and extend it to add the comparator functions for measuring columns and the value ones for the projection columns. Just to make the responsibility very clear on how these APIs are used. Not a must do, but for someone just looking at this class without looking at the usage the APIs are a bit confusing

e.g. getComparable and getValue return the same underlying arrays

Reply:
Good point. Refactored to 2 classes.